lib/taiko.js

```diff
@@ -2068,9 +2068,53 @@ for(const func in module.exports){
             return await realFuncs[func].apply(this, arguments);
         };
 }
+/**
+ * Mimics browser back button click functionality.
+ *
+ * @example
+ * goBack()
+ *
+ * @param {Object} options
+ * @returns {Promise<Object>} - Object with the description of the action performed.
+ */
+
+module.exports.goBack = async (options = {timeout: 30000}) =>{
+  _go(-1, options);
+  return { description: 'Performed clicking on browser back button' };
+};
+
+/**
+ * Mimics browser forward button click functionality.
+ *
+ * @example
+ * goForward()
+ *
+ * @param {Object} options
+ * @returns {Promise<Object>} - Object with the description of the action performed.
+ */
+
+module.exports.goForward = async (options = {timeout: 30000}) =>{
+  _go(+1, options);
+  return { description: 'Performed clicking on browser forward button' };
+};
+
+const _go = async (delta, options = {timeout: 30000}) =>{
+  const history = await page.getNavigationHistory();
+  const entry = history.entries[history.currentIndex + delta];
+
+  if (!entry)
+    return null;
+
+  const [response] = await Promise.all([
+    await waitForNavigation(options.timeout).catch(handleTimeout(options.timeout)),
```

Comment:
waitForNavigation takes promises to wait for, [Reference](https://github.com/getgauge/taiko/blob/32dfa0c8f05f56b2eeb31acce19b1a62051edfcc/lib/taiko.js#L256)
without promises it will not wait for the desired events currently

Reply:
worked on the comment